stellar-contract-env-host/src/host.rs

```diff
@@ -1332,99 +1336,150 @@ impl CheckedEnv for Host {
+        let res = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok(a.rem(b)))?)
+        })?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_and(&self, x: Object, y: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok(a.bitand(b)))?)
+        })?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_or(&self, x: Object, y: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok(a.bitor(b)))?)
+        })?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_xor(&self, x: Object, y: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok(a.bitxor(b)))?)
+        })?;
+        Ok(self.add_host_object(res)?.into())
     }
 
-    fn bigint_shl(&self, x: Object, y: RawVal) -> Result<Object, HostError> {
-        todo!()
+    fn bigint_shl(&self, x: Object, y: i64) -> Result<Object, HostError> {
+        let res = self.visit_obj(x, |a: &BigInt| Ok(a.shl(y)))?;
+        Ok(self.add_host_object(res)?.into())
     }
 
-    fn bigint_shr(&self, x: Object, y: RawVal) -> Result<Object, HostError> {
-        todo!()
+    fn bigint_shr(&self, x: Object, y: i64) -> Result<Object, HostError> {
+        let res = self.visit_obj(x, |a: &BigInt| Ok(a.shr(y)))?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_cmp(&self, x: Object, y: Object) -> Result<RawVal, HostError> {
-        todo!()
+        let res: RawVal = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok((a.cmp(b) as i32).into()))?)
+        })?;
+        Ok(res)
     }
 
     fn bigint_is_zero(&self, x: Object) -> Result<RawVal, HostError> {
-        todo!()
+        let res: RawVal = self.visit_obj(x, |a: &BigInt| Ok(a.is_zero().into()))?;
+        Ok(res)
     }
 
     fn bigint_neg(&self, x: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| Ok(a.neg()))?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_not(&self, x: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| Ok(a.not()))?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_gcd(&self, x: Object, y: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok(a.gcd(b)))?)
+        })?;
+        Ok(self.add_host_object(res)?.into())
     }
 
     fn bigint_lcm(&self, x: Object, y: Object) -> Result<Object, HostError> {
-        todo!()
+        let res = self.visit_obj(x, |a: &BigInt| {
+            Ok(self.visit_obj(y, |b: &BigInt| Ok(a.lcm(b)))?)
```

Comment:
Also looks like this handles 0 correctly, but we should be sure to test this (same thing about opening an issue).

Reply:
Tests now covers it.